### PR TITLE
Fix updating solution fallback options

### PR DESCRIPTION
**Pull request by tmat**
Fixes https://github.com/dotnet/roslyn/issues/75393, https://dev.azure.com/devdiv/DevDiv/_workitems/edit/2274440

### CODE REVIEW

**Comment by tmat** [comment]
@sharwell ptal
@arkalyanms

File: src/LanguageServer/Protocol/Features/Options/SolutionAnalyzerConfigOptionsUpdater.cs [hunk 1]
```diff
@@ -70,14 +70,17 @@ Solution UpdateOptions(Solution oldSolution)
                             continue;
                         }
 
-                        lazyBuilder ??= ImmutableDictionary.CreateBuilder<string, string>(AnalyzerConfigOptions.KeyComparer);
-
-                        // copy existing option values:
-                        foreach (var oldKey in languageOptions.Keys)
+                        if (lazyBuilder == null)
                         {
-                            if (languageOptions.TryGetValue(oldKey, out var oldValue))
```
[the remark this answers is not in the document]
Comment (JoeRobich) [reply src/LanguageServer/Protocol/Features/Options/SolutionAnalyzerConfigOptionsUpdater.cs:78]: Should we go ahead and `break` after copying options to the builder?

[the remark this answers is not in the document]
Comment (JoeRobich) [reply src/LanguageServer/Protocol/Features/Options/SolutionAnalyzerConfigOptionsUpdater.cs:78]: Or actually, the loop through `args.ChangedOptions` could become an `.Any()`

[the remark this answers is not in the document]
Comment (tmat) [reply src/LanguageServer/Protocol/Features/Options/SolutionAnalyzerConfigOptionsUpdater.cs:78]: The intention is that we copy all existing values and overwrite changed options with new values.

[the remark this answers is not in the document]
Comment (JoeRobich) [reply src/LanguageServer/Protocol/Features/Options/SolutionAnalyzerConfigOptionsUpdater.cs:78]: yep, got it now.

**Review by JoeRobich — APPROVED** [review]

**Review by sharwell — APPROVED** [review]

**Pull request merged**